### ISSUE TITLE
microsoft graph request throws an error but no message or status code is being returned

### ISSUE THREAD

**Issue by gabriel-tohos** [OfficeDev/microsoft-teams-library-js#1531]
A couple of months back I've implemented a feature in order to fetch the `teamsAppID` from the `AppCatalog` using microsoft graph.

Now, this feature works only when I run the bot locally. When deployed, we catch an error but no message or status code is being displayed (as seen in the screenshot below).

![image-20230110-095417](https://user-images.githubusercontent.com/122611075/212320598-557a3cb5-e23b-40ba-a014-91a6330fc925.png)

I've checked the env vars, token, baseURL, externalId, app level permissions and all but everything seems to be fine.

This is how I've implemented the above-mention function:

```typescript
    private readonly httpClient: AxiosInstance
    
    constructor(token: string) {
        this.httpClient = axios.create({
            baseURL: MICROSOFT_GRAPH_URL,
            headers: {'Authorization': `Bearer ${token}`}
        })
    }
    
    async getTeamsAppID(externalID: string): Promise<string> {
      try {
            const res = await this.httpClient.get<{ value: [TeamsApp] }>(`/appCatalogs/teamsApps?$filter=externalId eq '${externalID}'`)
    
            const teamsApps = res.data.value
    
            if (teamsApps.length > 0) {
                return teamsApps[0].id!
            }
      } catch (err: any) {
          logger.error("PBMSTEAMSBOT__MsGraphError__getTeamsAppID", {error: err})
      
          const e = err as AxiosError
          logger.error("PBMSTEAMSBOT__MsGraphError", {
              error: {
                  status: e?.response?.status,
                  code: e?.code,
                  data: e?.response?.data
              }
          })
      }
    
     return ""
}
```

Is there anyone else facing the same issue?

Versions:
    axios: 1.2.0
    microsoft-graph-types: 2.25.0

**Comment by ChetanSharma-msft** [comment]
Hello @gabriel-tohos - Thanks for reporting your issue.
We will look into it and let you know the updates.

**Comment by Prasad-MSFT** [comment]
@gabriel-tohos - Could you please provide the request id and time stamp when the request is failing, so that we can investigate at our end?

**Comment by gabriel-tohos** [comment]
@Prasad-MSFT Since we're not receiving any error details in the catch block, I'm not able to access the request id. The latest error happened on our end at `2023-01-16T12:40:08.464Z` UTC time.

**Comment by Prasad-MSFT** [comment]
@gabriel-tohos - Request id is required to investigate the issue further. Could you please share your console log as well?

**Comment by gabriel-tohos** [comment]
It seems that the data we're receiving it's somehow encoded. The first one is a simple `console.log()` of the response data, the other one is the `json.stringify()`-ed value. 

![image](https://user-images.githubusercontent.com/122611075/213118652-32b78669-2d2c-4f97-ad6a-dfa7fc9e38eb.png)

**Comment by Prasad-MSFT** [comment]
@gabriel-tohos - Could you please try to decode that and check?

**Comment by gabriel-tohos** [comment]
It seems that the axios version was at fault. In v1.2.0 the `Accept-Encoding` header was set to `gzip` by default. This issue was fixed in v1.2.1 by setting it to `application/json`, so everything is working as expected now. That you so much for your time and involvement, and sorry for this misunderstanding.
